core/src/main/scala/org/apache/spark/sql/delta/PreprocessTableWithDVs.scala

```diff
@@ -171,16 +169,24 @@ object ScanWithDeletionVectors {
 
   private def createBroadcastDVMap(
       spark: SparkSession,
-      tahoeFileIndex: TahoeFileIndex): Broadcast[Map[URI, DeletionVectorDescriptor]] = {
+      tahoeFileIndex: TahoeFileIndex
+  ): Broadcast[Map[URI, DeletionVectorDescriptorWithFilterType]] = {
+    val filterTypes = tahoeFileIndex.rowIndexFilters.getOrElse(Map.empty)
+
     // Given there is no way to find the final filters, just select all files in the
     // file index and create the DV map.
-    val filesWithDVs =
-      tahoeFileIndex.matchingFiles(Seq(TrueLiteral), Seq(TrueLiteral))
-        .filter(_.deletionVector != null)
-    val filePathToDVMap = filesWithDVs
-      .map(x =>
-         absolutePath(tahoeFileIndex.path.toString, x.path).toUri -> x.deletionVector)
-      .toMap
+    val filesWithDVs = tahoeFileIndex
+      .matchingFiles(partitionFilters = Seq(TrueLiteral), dataFilters = Seq(TrueLiteral))
+      .filter(_.deletionVector != null)
+    val filePathToDVMap = filesWithDVs.map { addFile =>
+      val key = absolutePath(tahoeFileIndex.path.toString, addFile.path).toUri
+      val filterType =
+        filterTypes.getOrElse(addFile.path, RowIndexFilterType.IF_CONTAINED)
```

Comment:
is the look by `String` works? I remember we need to use the `.toUri.toString` in order to have the unique and complete path.

@larsk-db please check if this is ok.

Reply:
We do lookups using `URI` in `DeltaParquetFileFormat`. IMO it's more robust than translating them into strings.